src/main/java/io/aiven/kafka/connect/opensearch/BulkProcessor.java

```diff
@@ -367,14 +383,18 @@ private BulkResponse execute() throws Exception {
             return callWithRetry("bulk processing", () -> {
                 try {
                     final var response =
-                            client.bulk(new BulkRequest().add(batch), RequestOptions.DEFAULT);
+                            client.bulk(new BulkRequest().add(
+                                            batch.stream().map(DocWriteRequestWrapper::getDocWriteRequests)
+                                                    .collect(Collectors.toList())),
+                                    RequestOptions.DEFAULT);
                     if (!response.hasFailures()) {
                         // We only logged failures, so log the success immediately after a failure ...
                         LOGGER.debug("Completed batch {} of {} records", batchId, batch.size());
                         return response;
                     }
                     for (final var itemResponse : response.getItems()) {
                         if (itemResponse.isFailed()) {
+                            reportMalformedRecord(itemResponse);
```

Comment:
Changed the logic to report to DLQ only after all retry attempts are completed. 
This will happen only in case DLQ is configured (otherwise reporter will be null and reportMalformedRecords method will return with doing nothing).
Note that in case of an "aborted" failure, the records will be written to DLQ but the exception will be thrown as before, making the connector to stop.
In case a response was returned from OpenSearch, the error reported to DLQ will contain the rest status, action id and error message (as suggested). In case an exception was thrown (no response from Opensearch), it will be reported as-is.

Reply:
@C0urante 
Invoking reporter::report doesn't cause the task to fail, it just writes to the dlq.
If the task should fail according to its configuration (or according to the error - for example IOException), it will fail after reporting to dlq (in case dlq is configured). 
If it's configured to ignore failures such as version conflicts, it will report to dlq and continue the task without failing.
For my use case it sounds reasonable...am I missing something?
There are some tests I've added to BulkProcessorTest that might clarify this behavior

Reply:
@nataliezeller1 Invoking `reporter::report` actually can cause a task to fail; see more in [KIP-610](https://cwiki.apache.org/confluence/display/KAFKA/KIP-610%3A+Error+Reporting+in+Sink+Connectors#KIP610:ErrorReportinginSinkConnectors-ErrorsTolerance), which introduced this feature:

> The Errant Record Reporter will adhere to the existing DLQ error tolerance functionality. For example, if `errors.tolerance` is set to `ALL`, all errors will be tolerated; if the property is set to `NONE`, then the Errant Record Reporter will throw an exception detailing that the tolerance has been exceeded. Even if the developer chooses to catch and swallow any of these exceptions thrown during `report(...)`, the task is guaranteed to be killed following the completion of `put(...)`.

Reply:
@C0urante Thanks, I see your point.
I've changed the logic to report to dlq only in these cases:
1. There was a failing error
2. errors.tolerance is ALL and an error occurred (including error types that are configured to ignore or warn)

I've also added a warning to alert on cases where the connector is configured with a dlq and error.tolerance=none, but behavior.on.version.conflict or behavior.on.malformed.documents are set to ignore/warn. 
In these cases, the record will not be reported to dlq in order to avoid failing the task.

Reply:
@nataliezeller1 Thanks, glad we're on the same page.

I wonder if it's necessary to check the `errors.tolerance` property at all? I'm a little worried about that kind of logic in case the Kafka Connect framework is expanded and a third (or fourth, or fifth, etc.) possible value for that property is added.

I really think it'd be best to only invoke the reporter on cases where we would otherwise fail the connector, and not for issues that the user has explicitly configured the connector to ignore.

If users want to report everything to the DLQ, they could do this by setting `errors.tolerance` to `all` in the connector config, and setting the `behavior.on.<malforned.docs|version.conflict>` properties to `fail` (or leaving them unset, since `fail` is the default for them).

If we want to get really fancy, we could add a new `report` value for those properties that 1) Forces an assertion on task startup that an `ErrantRecordReporter` is available (and fails the task otherwise) and 2) causes failures related to each specific property to be passed to the reporter. This could be a little easier for new users to grok since it doesn't imply that the connector will "fail" on those errors, but it could also be overkill.

Reply:
@C0urante 
In our use case, we want to move invalid events (ones that fail to be indexed due to incompatible schema) to dlq, without failing the task. 
Adding the "report" value can answer our needs, as long as it suggests that we'll report to dlq in any case of failure (without checking the errors.tolerance property), so the task might be failed in the reporter in case errors.tolerance is none (in our case it will be "all"). I can make this change if we are aligned on the solution now :-)

Reply:
> Adding the "report" value can answer our needs, as long as it suggests that we'll report to dlq in any case of failure (without checking the errors.tolerance property)

Sounds good to me! Looking forward to the next patch; hoping we can get this in by the end of the week. Thanks for bearing with me on the design discussion!

Reply:
Hi @C0urante 
My last commit contains the changes we've discussed - 
Now we'll report to DLQ only when an error related to 'behavior.on.version.conflict'/'behavior.on.malformed.documents' occurs and "report" is configured for the respective property. 
The connector will fail to start if "report" is configured for one of these properties but no DLQ topic is configured.
Sorry for the delay, hope it looks OK now :)